File: java-vfs/src/main/java/emg/java/vfs/files/File.java
package emg.java.vfs.files;

public class File extends DirEntry {
    private final String contents;

    public File(String parentPath, String name, String contents) {
        super(parentPath, name);
        this.contents = contents;
    }

    public static File empty(String parentPath, String name) {
        return new File(parentPath, name, "");
    }

    @Override
    public File asFile() {
        return this;
    }

    @Override
    public String getType() {
        return "File";
    }

    @Override
    public boolean isFile() {
        return true;
    }

    public File appendContents(String newContents) {
        return setContents(contents + "\n" + newContents);
    }

    public File setContents(String newContents) {
        return new File(parentPath, name, newContents);
    }

    public String contents() {
        return contents;
    }
}

File: kotlin-vfs/src/main/kotlin/emg/kotlin/vfs/files/File.kt
package emg.kotlin.vfs.files

class File(parentPath: String, name: String, val contents: String = "") : DirEntry(parentPath, name) {
    companion object {
        fun empty(parentPath: String, name: String): File {
            return File(parentPath, name)
        }
    }

    override fun getType(): String = "File"

    override fun asFile(): File = this

    override fun isFile(): Boolean = true
    fun appendContents(newContents: String): File {
        return setContents(contents + "\n" + newContents)
    }

    fun setContents(newContents: String): File {
        return File(parentPath, name, newContents)
    }
}